nupic/utils.py

```diff
@@ -114,3 +125,48 @@ def write(self, proto):
     proto.windowSize = self.windowSize
     proto.slidingWindow = self.slidingWindow
     proto.total = self.total
+
+###########################################################################
+class CircularBuffer(object):
+  """
+  implementation of a fixed size constant random access circular buffer
+  #TODO can this be merged with MovingAverage somehow?
```

Comment:
Let's move this to a separate PR. The implementation of this class makes it look totally useless to me so I must not be understanding it correctly. We should take a look at the usage and probably replace with a regular Python list or `collections.deque`